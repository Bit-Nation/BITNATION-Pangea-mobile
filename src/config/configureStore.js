// @flow

import { createStore, applyMiddleware, compose, type Store } from 'redux';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import createSagaMonitor from '@clarketm/saga-monitor';
import reducers from '../reducers';
import rootSaga from '../sagas';

/**
 * @desc Configures a Redux store.
 * @return {Store} Created store object.
 */
export default function configureStore(): Store {
  const sagaMonitor = createSagaMonitor({
    level: 'log',
    actionDispatch: true,
  });
  const sagaMiddleware = createSagaMiddleware({ sagaMonitor });

  const enhancer = compose(
    applyMiddleware(sagaMiddleware),
    applyMiddleware(logger),
  );

  const store = createStore(reducers, enhancer);
  sagaMiddleware.run(rootSaga);
  return store;
}
