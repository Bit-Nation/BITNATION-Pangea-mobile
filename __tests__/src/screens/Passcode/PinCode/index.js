import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import PinCodeScreen from '../../../../../src/screens/Passcode/PinCode';
import navigatorMock from '../../../../../__mocks__/Navigator';

describe('PinCodeScreen', () => {
  let pinCodeScreen;

  const initialStateMock = {};
  const storeMock = configureStore([]);
  const propsMock = {
    navigator: navigatorMock,
    shouldShowCancel: false,
    onSubmit: jest.fn(),
    onCancel: jest.fn(),
  };

  beforeEach(() => {
    pinCodeScreen = shallow(<PinCodeScreen {...propsMock} store={storeMock(initialStateMock)} />);
  });

  test('Rendering', () => {
    expect(pinCodeScreen).toMatchSnapshot();
  });
});
