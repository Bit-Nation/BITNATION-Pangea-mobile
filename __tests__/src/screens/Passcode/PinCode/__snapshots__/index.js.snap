// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinCodeScreen Rendering 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PinCodeScreen
    navigator={
      Object {
        "pop": [MockFunction],
        "push": [MockFunction],
        "setButtons": [MockFunction] {
          "calls": Array [
            Array [
              Object {},
            ],
          ],
        },
        "setOnNavigatorEvent": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
        },
      }
    }
    onCancel={[MockFunction]}
    onSubmit={[MockFunction]}
    shouldShowCancel={false}
    shouldShowForget={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#58595B",
              "fontFamily": "Roboto",
              "fontSize": 17,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 22,
            }
          }
        />,
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#F5A623",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            keyboardType="number-pad"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            returnKeyType="done"
            secureTextEntry={true}
            style={
              Object {
                "color": "#58595B",
                "fontSize": 16,
                "height": 50,
                "marginLeft": 30,
                "marginRight": 30,
                "marginTop": 16,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>,
        false,
      ],
      "style": Object {
        "alignContent": "flex-start",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginLeft": 8,
        "marginRight": 8,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": undefined,
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#58595B",
            "fontFamily": "Roboto",
            "fontSize": 17,
            "fontWeight": "bold",
            "letterSpacing": 0,
            "lineHeight": 22,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TextInput
            allowFontScaling={true}
            autoFocus={true}
            keyboardType="number-pad"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            returnKeyType="done"
            secureTextEntry={true}
            style={
              Object {
                "color": "#58595B",
                "fontSize": 16,
                "height": 50,
                "marginLeft": 30,
                "marginRight": 30,
                "marginTop": 16,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />,
          "style": Object {
            "borderBottomWidth": 1,
            "borderColor": "#F5A623",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoFocus": true,
            "keyboardType": "number-pad",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "returnKeyType": "done",
            "secureTextEntry": true,
            "style": Object {
              "color": "#58595B",
              "fontSize": 16,
              "height": 50,
              "marginLeft": 30,
              "marginRight": 30,
              "marginTop": 16,
              "paddingTop": 2,
              "textAlign": "center",
            },
            "underlineColorAndroid": "transparent",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#58595B",
                "fontFamily": "Roboto",
                "fontSize": 17,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 22,
              }
            }
          />,
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#F5A623",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoFocus={true}
              keyboardType="number-pad"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              returnKeyType="done"
              secureTextEntry={true}
              style={
                Object {
                  "color": "#58595B",
                  "fontSize": 16,
                  "height": 50,
                  "marginLeft": 30,
                  "marginRight": 30,
                  "marginTop": 16,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>,
          false,
        ],
        "style": Object {
          "alignContent": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "marginLeft": 8,
          "marginRight": 8,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#58595B",
              "fontFamily": "Roboto",
              "fontSize": 17,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 22,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TextInput
              allowFontScaling={true}
              autoFocus={true}
              keyboardType="number-pad"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              returnKeyType="done"
              secureTextEntry={true}
              style={
                Object {
                  "color": "#58595B",
                  "fontSize": 16,
                  "height": 50,
                  "marginLeft": 30,
                  "marginRight": 30,
                  "marginTop": 16,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />,
            "style": Object {
              "borderBottomWidth": 1,
              "borderColor": "#F5A623",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoFocus": true,
              "keyboardType": "number-pad",
              "onChangeText": [Function],
              "onSubmitEditing": [Function],
              "returnKeyType": "done",
              "secureTextEntry": true,
              "style": Object {
                "color": "#58595B",
                "fontSize": 16,
                "height": 50,
                "marginLeft": 30,
                "marginRight": 30,
                "marginTop": 16,
                "paddingTop": 2,
                "textAlign": "center",
              },
              "underlineColorAndroid": "transparent",
              "value": "",
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
